mlir/include/air/Util/Util.h

```diff
@@ -38,6 +38,13 @@ scf::ParallelOp getParallelRegionInitValsOwner(Operation *op, Value val);
 HerdOp getHerdArgOwner(Value val);
 // Get the parent air.hierarchy op of a tile id
 HierarchyInterface getHierarchyArgOwner(Value val);
+// Get the scf parent op from scf.yield op
+template <typename T> Operation *getScfParentOpFromYieldOp(scf::YieldOp op) {
+  if (auto scfop = dyn_cast<T>(op->getParentOp())) {
+    return scfop.getOperation();
+  }
+  return nullptr;
+}
```

Comment:
If you make this return type `T` then the user doesn't need to cast the returned value. If it also took an `Operation*` instead of a `scf::YieldOp` it would be general and a new form of `Operation::getParentOfType`.  But if it were used more generally then `dyn_cast_if_present` should be used because `getParentOp` can return `nullptr`

Reply:
Thanks for the suggestions, Jeff! I've modified this function accordingly.